src/audiovideocontroller/AudioVideoController.ts

```diff
@@ -42,12 +42,12 @@ export default interface AudioVideoController extends AudioVideoControllerFacade
   /**
    * Restarts the peer connection and/or the session.
    */
-  reconnect(status: MeetingSessionStatus): boolean;
+  reconnect(status: MeetingSessionStatus, errorMessage?: string | null): boolean;
 
   /**
    * Handles the meeting session status and returns true if it will restart the session.
    */
-  handleMeetingSessionStatus(status: MeetingSessionStatus): boolean;
+  handleMeetingSessionStatus(status: MeetingSessionStatus, errorMessage?: string | null): boolean;
```

Comment:
- Nice idea. Instead of a string, how about taking an Error object? In a place where you log, you can access `error.message` (and more properties such as `error.name` and `error.stack` later)

- Recommend removing `?`. Although we are using an optional argument (`?`) in several places, we prefer a required argument over `?`.  In case of no error from the caller, they can pass `null`.

  There will just be three more calls in `MonitorTask`, one of which has an error object too!

Reply:
1) Yep, passing Error is more flexible, will change in the next revision. 2) Will remove optional argument and add error in MonitorTask